### ISSUE TITLE
[AVM Module Issue]: Should be able to add a Sql Server Database to an existing Sql Server

### ISSUE THREAD

**Issue by gnfsdybvik** [Azure/bicep-registry-modules#3232]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/sql/server

### (Optional) Module Version

0.7.0

### Description

I can't see that it is possible to add a sqldb resource (Microsoft.Sql/servers/databases) to an existing br/public:avm/res/sql/server instance. It always have to be added at the same time as the server as a parameter to the server bicep module. We have a resource group where we store data for several services. We have a bicep script for the common infrastructure (like the sql server etc.) and then we have a separate bicep scripts for each service which add resources to that resource group later on. As of now the common bicep script need to know all the databases and add them to the server as you create the server. **Can we get the possibility to add the databases separately?**

It could be as easy as making the [bicep script for the database](https://github.com/Azure/bicep-registry-modules/blob/main/avm/res/sql/server/database/main.bicep) available in br/public.

Sidenote: This may apply to other resource types which as well (e.g. blob containers in a Storage Account?). Is there a design reason behind this way of constructing the bicep modules?

**Comment by emichellecarter** [comment]
Will look into this.

**Comment by peterbud** [comment]
Please see this FAQ: https://azure.github.io/Azure-Verified-Modules/faq/#can-i-call-a-bicep-child-module-directly-eg-can-i-update-or-add-a-secret-in-an-existing-key-vault-or-a-route-in-an-existing-route-table

TLDR: With the current module registry implementation it is not possible.

**Comment by timkatje** [comment]
Can we get a re-look at this request now that we have a concept of sub/child-modules in AVM? For instance, the Key Vault module now has accessible sub-modules for access policy, keys and secrets.

There are many use cases for being able to deploy a new database to existing servers, whether AVM-provisioned or not.  Similarly, we should be able to provision new elastic pools into existing servers.

I would really love to see the SQL team take on exposing the sub-modules of their project to end users, or some hint that this is already in the works.

**Comment by peterbud** [comment]
@timkatje Have you tried the new database child module?
